build.psm1

```diff
@@ -2997,7 +2997,13 @@ function New-TestPackage
         Write-Verbose -Message "Creating destination folder: $Destination"
     }
 
-    $packageRoot = Join-Path $env:TEMP ('TestPackage-' + (new-guid))
+    $rootFolder = $env:TEMP
+
+    if (-not $rootFolder -and $env:TF_BUILD) {
+        $rootFolder = $env:AGENT_WORKFOLDER
+    }
+
+    $packageRoot = Join-Path $rootFolder ('TestPackage-' + (new-guid))
```

Comment:
Is it possible to have both `$env:Temp` and `$env:TF_BUILD` be null?  Maybe we should have a `Test-Path` here and throw an error message ... "package path cannot be found"?

Reply:
$env:TEMP should be set in most cases. VSTS macOS hosts seem to be the only place where it is not set. $env:TF_BUILD is always set in VSTS builds.